shapeshift/web
# feat: arbitrary protocol fee balance check

## Description

Validates a trade for arbitrary protocol fees, rather than only buy/sell/fee assets. This is in preparation for multi-hop trades where a single trade may contain protocol fees in intermediary assets.

## Pull Request Type


## Issue (if applicable)

NA

## Risk

Low risk of UI incorrectly disallowing or allowing trades with insufficient balance for protocol fees.

## Testing

- Check trades with multiple protocol fees still work
- Check you cant submit a trade with insufficient balance for protocol fees

### Engineering

See above

### Operations

See above

## Screenshots (if applicable)
